RxSwift/Disposables/CompositeDisposable.swift

```diff
@@ -30,18 +30,10 @@ public class CompositeDisposable : DisposeBase, Disposable, Cancelable {
     /**
      Initializes a new instance of composite disposable with the specified number of disposables.
     */
-    public init(_ disposable1: Disposable, _ disposable2: Disposable) {
-        _disposables!.insert(disposable1)
-        _disposables!.insert(disposable2)
-    }
-    
-    /**
-     Initializes a new instance of composite disposable with the specified number of disposables.
-    */
-    public init(_ disposable1: Disposable, _ disposable2: Disposable, _ disposable3: Disposable) {
-        _disposables!.insert(disposable1)
-        _disposables!.insert(disposable2)
-        _disposables!.insert(disposable3)
+    public init(_ disposables: Disposable...) {
```

Comment:
Hi @larryonoff ,

I am aware of variadic arguments, but those overloads where there as an optimization so there is no need to create a variadic argument array.

I see your point, and we are really missing a general version. 

Could you just add a version with 4 arguments + variadic rest to make sure we are using optimized version up to 3 arguments.